wippersnapper-config-builder.js

```diff
@@ -1578,22 +1583,6 @@ function saveModalData() {
             // Remove any components using this multiplexer
             appState.selectedComponents = appState.selectedComponents.filter(c =>
                 !(c.i2cMuxAddress && c.i2cMuxAddress === componentConfig.i2cDeviceAddress));
-        } else {
-            // Add data types for non-multiplexer components
-            const dataTypeCheckboxes = document.querySelectorAll('input[name="data-type"]:checked');
-            if (dataTypeCheckboxes.length > 0) {
-                componentConfig.i2cDeviceSensorTypes = Array.from(dataTypeCheckboxes).map(checkbox => {
-                    try {
-                        return { type: JSON.parse(checkbox.value) };
-                    } catch (e) {
-                        return { type: checkbox.value };
-                    }
-                });
-            } else {
-                validationError = true;
-                alert('Please select at least one data type for the I2C component.');
-                return false;
-            }
```

Comment:
Removing this validation for i2c components is not great.
 Instead it could be qualified with isGPS, thereby allowing the normal datatype validation to require at least one subcomponent (sensor type) selected when adding a component.

Reply:
Fixed